docs/guide/grouping.md

```diff
@@ -19,3 +19,127 @@ There are 3 table features that can reorder columns, which happen in the followi
 2. Manual [Column Ordering](../column-ordering) - A manually specified column order is applied.
 3. **Grouping** - If grouping is enabled, a grouping state is active, and `tableOptions.groupedColumnMode` is set to `'reorder' | 'remove'`, then the grouped columns are reordered to the start of the column flow.
+
+Grouping in TanStack table is a feature that applies to columns and allows you to categorize and organize the table rows based on specific columns. This can be useful in cases where you have a large amount of data and you want to group them together based on certain criteria.
+
+To use the grouping feature, you will need to use the grouped row model. This model is responsible for grouping the rows based on the grouping state.
+
+```tsx
+import {getGroupedRowModel} from '@tanstack/react-table'
+
+const table = useReactTable({
+  // other options...
+  getGroupedRowModel: getGroupedRowModel(),
+})
+```
+
+When grouping state is active, the table will add matching rows as subRows to the grouped row. The grouped row will be added to the table rows at the same index as the first matching row. The matching rows will be removed from the table rows.
+To allow the user to expand and collapse the grouped rows, you can use the expanding feature.
+
+```tsx
+import {getGroupedRowModel, getExpandedRowModel} from '@tanstack/react-table'
+
+const table = useReactTable({
+  // other options...
+  getGroupedRowModel: getGroupedRowModel(),
+  getExpandedRowModel: getExpandedRowModel(),
+})
+```
+
+### Grouping state
+
+The grouping state is an array of strings, where each string is the ID of a column to group by. The order of the strings in the array determines the order of the grouping. For example, if the grouping state is ['column1', 'column2'], then the table will first group by column1, and then within each group, it will group by column2. You can control the grouping state using the setGrouping function:
+
+```tsx
+table.setGrouping(['column1', 'column2']);
+```
+
+You can also reset the grouping state to its initial state using the resetGrouping function:
+
+```tsx
+table.resetGrouping();
+```
+
+By default, when a column is grouped, it is moved to the start of the table. You can control this behavior using the groupedColumnMode option. If you set it to 'reorder', then the grouped columns will be moved to the start of the table. If you set it to 'remove', then the grouped columns will be removed from the table. If you set it to false, then the grouped columns will not be moved or removed.
+
+```tsx
+const table = useReactTable({
+  // other options...
+  groupedColumnMode: 'reorder',
+})
+```
+
+### Aggregations
+
+When rows are grouped, you can aggregate the data in the grouped rows by columns using the aggregationFn option. This is a string that is the ID of the aggregation function. You can define the aggregation functions using the aggregationFns option.
+
+```tsx
+const column = columnHelper.data('key', {
```

Comment:
I don't think this is real code either

Reply:
If you mean the createColumnHelper does not have a data api which I just checked and it doesn't. I copied that piece of code the grouping api doc.